src/middleware.ts

```diff
@@ -21,6 +23,15 @@ export async function middleware(req: NextRequest) {
     return NextResponse.redirect(new URL("/auth/login", req.url));
   }
 
+  if (
+    token &&
+    (token.profile as { firstName: string })?.firstName &&
+    token &&
+    (token.profile as { lastName: string })?.lastName
+  ) {
+    return NextResponse.redirect(new URL("/sessions", req.url));
+  }
+
```

Comment:
Are you having typescript errors? Check the message I sent about extending the DefaultJWT field with the profile field.

Also does this achieve the result we want? Remember
- If the user is navigating to a `/onboard` page AND they already have the fields then redirect them to `/sessions`
- If they are on any other page and they don't have the fields then redirect them to `/onboard/name`.